bindings/matrix-sdk-crypto-nodejs/src/requests.rs

```diff
@@ -224,43 +241,89 @@ impl KeysBackupRequest {
 }
 
 macro_rules! request {
-    ($request:ident from $ruma_request:ident maps fields $( $field:ident $( { $transformation:expr } )? ),+ $(,)? ) => {
-        impl TryFrom<(String, &$ruma_request)> for $request {
+    (
+        $destination_request:ident from $source_request:ident
```

Comment:
any chance to get some unit tests for this macro?

It all seems fine to me but I find macro code incredibly difficult to comprehend and follow and as bugs in there are notoriously badly communicated, it is very prone to fail in unpredicted ways ...

Reply:
The request types are tested in the various test suites, there isn't a single place where they are explicitly tested, mostly because the public API doesn't allow to build them directly.

The best tests here is the type system. If it compiles, it means types are correctly transformed from Ruma and `matrix-sdk-crypto` to `matrix-sdk-crypto-nodejs`. It's really a type conversion from Rust to Rust, there is no magic here, so the type checker really is our “test”.